app/src/main/java/com/plenart/organizeme/activities/MainActivity.kt

```diff
@@ -3,188 +3,95 @@ package com.plenart.organizeme.activities
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.navigation.NavController
+import androidx.navigation.fragment.NavHostFragment
+import androidx.navigation.ui.setupActionBarWithNavController
 import com.google.firebase.auth.FirebaseAuth
 import com.plenart.organizeme.R
-import com.plenart.organizeme.adapters.BoardItemsAdapter
 import com.plenart.organizeme.databinding.ActivityMainBinding
-import com.plenart.organizeme.databinding.AppBarMainBinding
-import com.plenart.organizeme.databinding.MainContentBinding
 import com.plenart.organizeme.firebase.Firestore
-import com.plenart.organizeme.interfaces.BoardItemClickInterface
-import com.plenart.organizeme.models.Board
-import com.plenart.organizeme.utils.Constants
-import de.hdodenhof.circleimageview.CircleImageView
+import com.plenart.organizeme.viewModels.MainActivityViewModel
 
-class MainActivity : BaseActivity(), NavigationView.OnNavigationItemSelectedListener {
+class MainActivity : AppCompatActivity() {
 
-    private lateinit var mainActivityBinding: ActivityMainBinding;
-    private lateinit var appBarMainBinding: AppBarMainBinding;
-    private lateinit var mainContentBinding: MainContentBinding;
-
-    private lateinit var mUserName: String;
+    private lateinit var binding: ActivityMainBinding
+    private val viewModel: MainActivityViewModel by viewModels()
 
+    private lateinit var toolbar: Toolbar
+    private lateinit var navController: NavController
 
     override fun onCreate(savedInstanceState: Bundle?) {
+        setTheme(R.style.AppTheme)
         super.onCreate(savedInstanceState)
-        mainActivityBinding = ActivityMainBinding.inflate(layoutInflater)
-        setContentView(mainActivityBinding.root)
-
-        setUpActionBar();
-        mainActivityBinding.navView.setNavigationItemSelectedListener(this);
-
-        Firestore().loadUserData(this,true);
 
-        appBarMainBinding.fabCreateBoard.setOnClickListener{
-            val intent = Intent(this,CreateBoardActivity::class.java);
-            intent.putExtra(Constants.NAME, mUserName);
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
 
-            startActivityForResult(intent, CREATE_BOARD_REQUEST_CODE);
-        }
-
-    }
+        toolbar = binding.toolbarMainActivity
+        setSupportActionBar(toolbar)
 
+        setupNavController()
+        setupActionBarWithNavController(navController)
 
-    private fun setUpActionBar(){
-        appBarMainBinding = mainActivityBinding.appBarMainIncluded;
-
-        setSupportActionBar(appBarMainBinding.toolbarMainActivity)
-        appBarMainBinding.toolbarMainActivity.setNavigationIcon(R.drawable.ic_action_navigation_menu);
-
-        appBarMainBinding.toolbarMainActivity.setNavigationOnClickListener {
-            toggleDrawer();
-        }
+        Handler(Looper.getMainLooper()).postDelayed({
```

Comment:
Let's not do this.

Move delays into viewModel and implement this uisng coroutines. 

Further more, regarding Splash screens

Whenever launcher activity starts, it will for a few ms/s show themes default window background - in most cases it will be white, depends on parent theme. Best way to handle splash is to create one activity which will be used as launcher and will have it's own theme with default background set to our splash image. When launcher activity is created you can then easly start other and finish splash